longhorndev/dev.go

```diff
@@ -239,6 +239,9 @@ func (d *LonghornDevice) PrepareUpgrade() error {
 }
 
 func (d *LonghornDevice) FinishUpgrade() (err error) {
+	d.Lock()
```

Comment:
The newly added code is basically duplicated of `Start()` without creating target etc. Can we call a common function from Start() instead?